src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/SvDiscoverFromLocalAssemblyContigAlignmentsSpark.java

```diff
@@ -272,44 +281,197 @@ public static AssemblyContigsClassifiedByAlignmentSignatures preprocess(final Sv
      * {@link AssemblyContigWithFineTunedAlignments.AlignmentSignatureBasicType#UNKNOWN}
      * currently DO NOT generate any VCF yet.
      */
-    public static void dispatchJobs(final JavaSparkContext ctx,
-                                    final AssemblyContigsClassifiedByAlignmentSignatures contigsByPossibleRawTypes,
-                                    final SvDiscoveryInputMetaData svDiscoveryInputMetaData,
-                                    final JavaRDD<GATKRead> assemblyRawAlignments,
-                                    final boolean writeSAMFiles) {
+    public static List<VariantContext> dispatchJobs(final JavaSparkContext ctx,
+                                                    final AssemblyContigsClassifiedByAlignmentSignatures contigsByPossibleRawTypes,
+                                                    final SvDiscoveryInputMetaData svDiscoveryInputMetaData,
+                                                    final JavaRDD<GATKRead> assemblyRawAlignments,
+                                                    final boolean writeSAMFiles) {
 
         final String outputPrefixWithSampleName = svDiscoveryInputMetaData.getOutputPath();
 
-        // TODO: 1/10/18 bring back read annotation, see ticket 4228
+        final List<VariantContext> simpleChimeraVariants =
+                extractSimpleVariants(contigsByPossibleRawTypes.simple, svDiscoveryInputMetaData, outputPrefixWithSampleName);
+        contigsByPossibleRawTypes.simple.unpersist(false);
+
+        final CpxAndReInterpretedSimpleVariants complexChimeraVariants =
+                extractCpxVariants(ctx, contigsByPossibleRawTypes.complex, svDiscoveryInputMetaData, assemblyRawAlignments, outputPrefixWithSampleName);
+        contigsByPossibleRawTypes.complex.unpersist(false);
+
+        if (writeSAMFiles) {
+            contigsByPossibleRawTypes.writeSAMfilesForUnknown(outputPrefixWithSampleName, assemblyRawAlignments,
+                    svDiscoveryInputMetaData.getSampleSpecificData().getHeaderBroadcast().getValue());
+        }
+        contigsByPossibleRawTypes.unknown.unpersist(false);
+
+        final List<VariantContext> inversions = extractInversions();// TODO: 6/29/18 placeholder
+
+        // merged output
+        final List<VariantContext> merged = new ArrayList<>(20_000); // estimated size
```

Comment:
You've already got these three collections as lists so why not just add up their sizes to get the capacity instead of guessing?